DOC: Small fixes for convolution kernel docs

svank opened the pull request:
### Description

A few small doc fixes:

- The docs for `astropy.convolution.AiryDisk2DKernel` say "this kernel is normalized to a peak value of 1," but the normalization actually ensures that the sum, not the peak value, is 1 (see below). This PR adjusts the text accordingly.

```>>> from astropy.convolution import AiryDisk2DKernel
>>> from astropy.convolution import AiryDisk2DKernel
>>> k = AiryDisk2DKernel(10)
>>> k.array.sum()
1.0
>>> k.array.max()
0.012127255578959403
```

- Adds `Moffat2DKernel` to the list of available kernels on the "Convolution Kernels" docs page, where it had been omitted.
- Corrects inconsistencies in the "See Also" sections of the docs for `Ring2DKernel` and `AiryDisk2DKernel`.

dhomeier commented on astropy/convolution/kernels.py:
```diff
@@ -697,7 +697,7 @@ class AiryDisk2DKernel(Kernel2D):
 
     This kernel models the diffraction pattern of a circular aperture. This
-    kernel is normalized to a peak value of 1.
+    kernel is normalized so that it sums to 1.
```
```suggestion
    kernel is normalized so that it integrates to 1.
```
Seems better for the mathematical property, even if it amounts to a numerical summation. And it is achieved by calling the `self.normalize([mode='integral'])` method.

dhomeier reviewed (commented):
Thanks for looking into this; just a small suggestion on the wording.
For a potentially larger one, any kernel should integrate to 1 to be used out of the box for convolution, and as far as I can see all others are in fact normalised the same way, just without any note of it in the docs. So perhaps that note on normalisation could be trimmed entirely (or added to all kernel docstrings...).

svank commented:
@dhomeier I agree on replacing "sums" with "integrates", though I'll wait to change that pending your other suggestion.

The automatic normalization of all kernels is mentioned in the [general "convolution kernels" docs](https://docs.astropy.org/en/stable/convolution/kernels.html#normalization), but it's a couple clicks away from the AiryDisk2DKernel page---I wouldn't have found it if not for my deeper dive prompted by deciding to fix the "peak value of 1" language.

I would be inclined to copy the brief note to all kernel docstrings. It's a short note that is very relevant for most uses, and while the automatic normalization makes sense for these tools, it's not behavior I would have been comfortable assuming without seeing it mentioned.

I'll add a commit to this PR in ~a week making these two changes unless I hear any opposing thoughts.

svank commented:
I've replaced "sums" with "integrates", and I've copied the normalization note to the other kernels so that this important aspect is clear at a glance. (I also fixed a couple of typos in the Box kernels' docs.)

I did notice that the RickerWavelet kernels are *not* normalized. I don't know enough about those kernels to know if that's a bug, though. (Either way, I did not modify those kernels' docs.)

nstarman commented:
@keflavich, squash before merge, or is this GTG?
